plugins/auth-backend/src/lib/passport/BackstageIdentityStrategy.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Request } from 'express';
+import { JWK, JWT, JWKS } from 'jose';
+import { BackstageIdentity } from '../../providers';
+import { IdentityClient } from '../../identity';
+import { PluginEndpointDiscovery } from '@backstage/backend-common';
+
+const Strategy = require('passport-strategy');
+// Using singleton keyStore instead of membership due to Passport wtf
+let keyStore: JWKS.KeyStore;
+let keyStoreUpdated: number;
+
+export class BackstageIdentityStrategy extends Strategy {
+  private readonly client: IdentityClient;
+  private readonly discovery: PluginEndpointDiscovery;
+
+  constructor(options: { discovery: PluginEndpointDiscovery }) {
+    super();
+    this.client = new IdentityClient({ discovery: options.discovery });
+    this.discovery = options.discovery;
+    this.name = 'backstage';
+  }
+
+  private async refreshKeyStore(rawJwtToken: string) {
```

Comment:
Very much going to need some solid test coverage of these pieces 😅 . Would be good if you could add at least a few simple tests and we can fill in some more if needed. Should be easier if we split apart the validation logic from the passport strategy as well

Reply:
Agreed. Wanted to wait for initial input that we wanted to go this way before spending too much time on test and documentation.